diff --git a/cmake/platform.cmake b/cmake/platform.cmake
index 428f5f076..32e917a7a 100644
--- a/cmake/platform.cmake
+++ b/cmake/platform.cmake
@@ -145,7 +145,7 @@ if(MSVC)
         # We don't want to optimize jit gemm kernels to reduce compile time
         append(CMAKE_CCXX_FLAGS "-Wno-overriding-t-option")
     endif()
-    if(DNNL_WITH_SYCL OR CMAKE_BASE_NAME STREQUAL "icx" OR CMAKE_BASE_NAME STREQUAL "icpx")
+    if(DNNL_WITH_SYCL OR CMAKE_BASE_NAME MATCHES ".*icx$" OR CMAKE_BASE_NAME MATCHES ".*icpx$")
         # Default fp-model in icx and dpcpp (unlike clang) may be precise or
         # fast=1 depending on the version.
         append(CMAKE_CCXX_FLAGS "/fp:precise")
@@ -155,7 +155,7 @@ elseif(UNIX OR MINGW)
         # XXX: Intel oneAPI DPC++ Compiler generates a lot of warnings
         append(CMAKE_CCXX_FLAGS "-w")
     endif()
-    if(DNNL_WITH_SYCL OR CMAKE_BASE_NAME STREQUAL "icx" OR CMAKE_BASE_NAME STREQUAL "icpx")
+    if(DNNL_WITH_SYCL OR CMAKE_BASE_NAME MATCHES ".*icx$" OR CMAKE_BASE_NAME MATCHES ".*icpx$")
         # Default fp-model in icx and dpcpp (unlike clang) may be precise or
         # fast=1 depending on the version.
         append(CMAKE_CCXX_FLAGS "-ffp-model=precise -fno-reciprocal-math")
